src/AspNetCore/Inc/aspnetcore_event.h

```diff
@@ -0,0 +1,551 @@
+            Event.pActivityGuid = NULL;
+            Event.pRelatedActivityGuid = NULL;
+            Event.dwTimeStamp = 0;
+            Event.dwFlags = HTTP_TRACE_EVENT_FLAG_STATIC_DESCRIPTIVE_FIELDS;
+    
+            // pActivityGuid, pRelatedActivityGuid, Timestamp to be filled in by IIS
+    
+            HTTP_TRACE_EVENT_ITEM Items[ 2 ];
+            Items[ 0 ].pszName = L"ContextId";
+            Items[ 0 ].dwDataType = HTTP_TRACE_TYPE_LPCGUID; // mof type (object)
+            Items[ 0 ].pbData = (PBYTE) pContextId;
+            Items[ 0 ].cbData = 16;
+            Items[ 0 ].pszDataDescription = NULL;
+            Items[ 1 ].pszName = L"AppDescription";
+            Items[ 1 ].dwDataType = HTTP_TRACE_TYPE_LPCWSTR; // mof type (string)
+            Items[ 1 ].pbData = (PBYTE) pAppDescription;
+            Items[ 1 ].cbData  = 
+                 ( Items[ 1 ].pbData == NULL )? 0 : ( sizeof(WCHAR) * (1 + (DWORD) wcslen( (PWSTR) Items[ 1 ].pbData  ) ) );
+            Items[ 1 ].pszDataDescription = NULL;
+            Event.pEventItems = Items;
+            pHttpTraceContext->RaiseTraceEvent( &Event );
+            return S_OK;
+        };
+    
+        static
+        BOOL
+        IsEnabled( 
+            IHttpTraceContext *  pHttpTraceContext )
+        // Check if tracing for this event is enabled
+        {
+            return WWWServerTraceProvider::CheckTracingEnabled( 
+                                 pHttpTraceContext,
+                                 WWWServerTraceProvider::ANCM,
+                                 4 ); //Verbosity
+        };
+    };
+    //
+    // Event: mof class name ANCMAppStartFail,
+    // Description: Start application failed
+    // EventTypeName: ANCM_START_APPLICATION_FAIL
+    // EventType: 2
+    // EventLevel: 2
+    //
+    
+    class ANCM_START_APPLICATION_FAIL
+    {
+    public:
+        static
+        HRESULT
+        RaiseEvent(
+            IHttpTraceContext * pHttpTraceContext,
+            LPCGUID    pContextId,
+            LPCWSTR     pFailureDescription
+        )
+        //
+        // Raise ANCM_START_APPLICATION_FAIL Event
+        //
+        {
+            HTTP_TRACE_EVENT Event;
+            Event.pProviderGuid = WWWServerTraceProvider::GetProviderGuid();
+            Event.dwArea =  WWWServerTraceProvider::ANCM;
+            Event.pAreaGuid = ANCMEvents::GetAreaGuid();
+            Event.dwEvent = 2;
+            Event.pszEventName = L"ANCM_START_APPLICATION_FAIL";
+            Event.dwEventVersion = 1;
+            Event.dwVerbosity = 2;
+            Event.cEventItems = 2;
+            Event.pActivityGuid = NULL;
+            Event.pRelatedActivityGuid = NULL;
+            Event.dwTimeStamp = 0;
+            Event.dwFlags = HTTP_TRACE_EVENT_FLAG_STATIC_DESCRIPTIVE_FIELDS;
+    
+            // pActivityGuid, pRelatedActivityGuid, Timestamp to be filled in by IIS
+    
+            HTTP_TRACE_EVENT_ITEM Items[ 2 ];
+            Items[ 0 ].pszName = L"ContextId";
```

Comment:
It seems there are only two event items, (ContextId and failure). Can we cache these and use them each time? I'm assuming that RaiseTraceEvent() does not modify the event passed in.